### ISSUE TITLE
Unable to load mapper plugin ../syncdata.so: plugin: not implemented

### ISSUE THREAD

**Issue by vasupal1996** [rwynn/monstache#303]
I'm getting this error when I try to run monstache with a plugin.

These are the steps I followed to build the plugin.
1. cloned monstache and git checkout v6.0.11 and go install
2. Created syncdata.go
3. `go build -buildmode=plugin -o syncdata.so syncdata.go`
4. copied syncdata.so file to mustache executable directory
5. running using `./monstache -f ../config-change-stream.toml -mapper-plugin-path ../syncdata.so`

Can anyone help me fix this?

**Comment by rwynn** [comment]
@vasupal1996 any chance you are running monstache on windows?  If so, the problem would be that golang only supports plugins on linux and osx.  If not, then it must be some other problem.

**Comment by vasupal1996** [comment]
I am running this locally on mac os x.

**Comment by rwynn** [comment]
When you perform the first step...

```
cloned monstache and git checkout v6.0.11 and go install
```

That should result in a monstache binary saved by default in `~/go/bin`.  Are you running the plugin against this monstache binary or some other?

Also, does your `syncdata.go` have `package main` at the top?  Here is an [example](https://github.com/rwynn/monstache-showcase/blob/master/plugin/plugin.go).

**Comment by leandono** [comment]
Hello, I'm getting the same problem on Mac OS running a plugin compiled against the binary downloaded from the releases page. But building monstache locally and running the plugin against the binary compiled the problem doesn't happens.
The only way to run monstache with a plugin is building the binary or it can works with any binary downloaded?

**Comment by rwynn** [comment]
@leandono, yes I see the same problem on mac os x.  I think it may have to do with how the binary for mac is cross compiled on linux for darwin.

To support the plugin on release binary on mac I may need to compile on mac os x or alter somehow the build script in the Makefile at https://github.com/rwynn/monstache/blob/v6.4.1/Makefile that gets run on Linux.  

For now if you want to run plugin on mac os x you can...

* Run `go install` in root monstache directory after git checkout of release tag .e.g. `v6.4.1` to produce `monstache` binary and then run plugin.so against that binary (like you mentioned), or...
* Use docker for mac to run a release image of monstache. e.g. `rwynn/monstache:6.4.1` against a plugin you build using the script at https://github.com/rwynn/monstache/tree/v6.4.1/docker/plugin

**Comment by shanedeeodonnell** [comment]
@rwynn I'm also on os x and getting the same error even after following the instructions above. Do you have any insight? Here's what I've done:

- `git clone` and `git checkout` of release tag v6.5.2
- `go install` in root of this new dir
- `cp monstache*` from GOPATH/bin to root of source dir from the first step
- `go build -buildmode=plugin -o myPlugin.so myPlugin.go` which generates myPlugin.so fine
-  executing `monstache -mapper-plugin-path myPlugin.so` gives me the `plugin: not implemented` error
-  executing `./monstache -mapper-plugin-path myPlugin.so` gives me the `Unable to load mapper plugin myPlugin.so: plugin.Open("myPlugin.so"): realpath failed` 

Monstache is a fantastic tool btw, thanks for all the work you've put into it.

**Comment by rwynn** [comment]
@shanedeeodonnell thanks.  It seems to be having trouble finding the .so.  Can you try with the full path to it like `-mapper-plugin-path /full/path/to/myPlugin.so`?

**Comment by shanedeeodonnell** [comment]
@rwynn so simple. Thank you, it's running as expected.

**Comment by maheswaranperfios** [comment]
Hey @rwynn I am also currently facing this issue. 
`ERROR 2025/07/11 08:33:38 Unable to load mapper plugin /app/plugin.so: plugin: not implemented`

I am using GO version - go version go1.24.4 linux/amd64
and monstache version 6.7.22

I am providing absolute full path of plugin in config.toml for keyword mapper-plugin-path 

Could you please help me with this.?

**Comment by maheswaranperfios** [comment]
Nevermind, it is resolved.I was downloading the binary from github instead of building it as you mentioned in the docs so my bad. After building the binary with go install and using it solved the issue. Hope it helps someone.

> Hey [@rwynn](https://github.com/rwynn) I am also currently facing this issue. `ERROR 2025/07/11 08:33:38 Unable to load mapper plugin /app/plugin.so: plugin: not implemented`
> 
> I am using GO version - go version go1.24.4 linux/amd64 and monstache version 6.7.22
> 
> I am providing absolute full path of plugin in config.toml for keyword mapper-plugin-path
> 
> Could you please help me with this.?
